### PR TITLE
swap HG/LG channels

### PR DESCRIPTION
**Pull request by peressounko**

### CODE REVIEW

File: Detectors/PHOS/reconstruction/src/AltroDecoder.cxx [hunk 1]
```diff
@@ -65,8 +65,14 @@ void AltroDecoder::readChannels(const std::vector<uint32_t>& buffer, CaloRawFitt
       if (currentword != 0) {
         LOG(ERROR) << "Channel header mark not found, header word " << currentword;
         short fec = header.mHardwareAddress >> 7 & 0xf; //try to extract FEE number from header
-        if (fec < 0 || fec > 13) {
-          fec = kGeneralSRUErr;
+        short branch = header.mHardwareAddress >> 11 & 0x1;
+        if (branch < 0 || branch > 1) {
+          fec = kGeneralTRUErr + kGeneralSRUErr;
```
Comment (shahor02) [inline Detectors/PHOS/reconstruction/src/AltroDecoder.cxx]: Result of anything & 0x1 is either 0 or 1, so the condition of this `if` is always false. What is its point?

File: Detectors/PHOS/reconstruction/src/AltroDecoder.cxx [hunk 2]
```diff
@@ -65,8 +65,14 @@ void AltroDecoder::readChannels(const std::vector<uint32_t>& buffer, CaloRawFitt
...
+        } else {
+          if (fec < 0 || fec > 13) {
+            fec = kGeneralSRUErr;
```
Comment (shahor02) [inline Detectors/PHOS/reconstruction/src/AltroDecoder.cxx]: also here, fec is limited to 0:15 ( = 0xf), so the <0 comparison is meaningless.

File: Detectors/PHOS/reconstruction/src/AltroDecoder.cxx [hunk 3]
```diff
@@ -77,8 +83,14 @@ void AltroDecoder::readChannels(const std::vector<uint32_t>& buffer, CaloRawFitt
...
-      if (fec < 0 || fec > 13) {
-        fec = kGeneralSRUErr;
+      short branch = header.mHardwareAddress >> 11 & 0x1;
+      if (branch < 0 || branch > 1) {
+        fec = kGeneralTRUErr + kGeneralSRUErr;
+      } else {
+        if (fec < 0 || fec > 13) {
+          fec = kGeneralSRUErr;
+        }
+        fec += kGeneralTRUErr * branch;
```
Comment (shahor02) [inline Detectors/PHOS/reconstruction/src/AltroDecoder.cxx]: same here and below.

**Review by shahor02 — CHANGES_REQUESTED** [review]

File: Detectors/PHOS/reconstruction/src/AltroDecoder.cxx (hunk 3 above)
Comment (peressounko) [reply Detectors/PHOS/reconstruction/src/AltroDecoder.cxx]: fixed

File: Detectors/PHOS/reconstruction/src/AltroDecoder.cxx (hunk 2 above)
Comment (peressounko) [reply Detectors/PHOS/reconstruction/src/AltroDecoder.cxx]: Rudiment. Fixed in new version

**Comment by peressounko** [comment]
dear @shahor02 , could you please approve? test build/AliceO2/O2/o2/macOS fail for reasons not PHOS-related.

**Review by shahor02 — APPROVED** [review]

**Pull request merged**